lib/javascript/BasicEvaluatedExpression.js

```diff
@@ -97,6 +114,86 @@ class BasicEvaluatedExpression {
 		return this.type === TypeTemplateString;
 	}
 
+	/**
+	 * Is expression a primitive or an object type value?
+	 * @returns {boolean | undefined} true: primitive type, false: object type, undefined: unknown/runtime-defined
+	 */
+	isPrimitiveType() {
+		switch (this.type) {
+			case TypeUndefined:
+			case TypeNull:
+			case TypeString:
+			case TypeNumber:
+			case TypeBoolean:
+			case TypeBigInt:
+			case TypeWrapped:
+			case TypeTemplateString:
+				return true;
+			case TypeRegExp:
+			case TypeArray:
+			case TypeConstArray:
+				return false;
+			default:
+				return undefined;
+		}
+	}
+
+	/**
+	 * Is expression a runtime or compile-time value?
+	 * @returns {boolean} true: compile time value, false: runtime value
+	 */
+	isCompileTimeValue() {
+		switch (this.type) {
+			case TypeUndefined:
+			case TypeNull:
+			case TypeString:
+			case TypeNumber:
+			case TypeBoolean:
+			case TypeRegExp:
+			case TypeConstArray:
+			case TypeBigInt:
+				return true;
+			case TypeWrapped:
+			case TypeIdentifier:
+			case TypeConditional:
+			case TypeArray:
+			case TypeTemplateString:
+			case TypeUnknown:
+				return false;
+			default:
+				return false;
```

Comment:
```suggestion
				return true;
			default:
				return false;
```
probably improves readability also..